Cubical/Codata/Conat/Properties.agda

```diff
@@ -39,37 +41,64 @@ force (∞+1≡∞ _) = suc ∞
 
 -- TODO: plus for conat, ∞ + ∞ ≡ ∞
 
-record _~_ (x y : Conat) : Set
-data _~′_ (x y : Conat′) : Set
-_~′′_ : Conat′ → Conat′ → Set
-zero  ~′′ zero  = Unit
-suc x ~′′ suc y = x ~ y
--- So impossible proofs are preserved
-x ~′′ y = x ≡ y
+module Bisimulation where
 
-record _~_ x y where
-  coinductive
-  field prove : force x ~′ force y
+  record _≈_ (x y : Conat) : Set
+  data _≈′_ (x y : Conat′) : Set
+  _≈′′_ : Conat′ → Conat′ → Set
+  -- zero  ≈′′ zero  = Unit
+  suc x ≈′′ suc y = x ≈ y
+  -- So impossible proofs are preserved
+  x ≈′′ y = x ≡ y
 
-data _~′_  x y where
-  con : x ~′′ y → x ~′ y
+  record _≈_ x y where
+    coinductive
+    field prove : force x ≈′ force y
 
-open _~_ public
+  data _≈′_  x y where
+    con : x ≈′′ y → x ≈′ y
 
-bisim : ∀ {x y} → x ~ y → x ≡ y
-force (bisim eq i) = bisim′ (prove eq) i
-  where
-  bisim′ : ∀ {x y} → x ~′ y → x ≡ y
-  bisim′ {zero} {zero}  (con tt) = refl
+  open _≈_ public
+
+  bisim : ∀ {x y} → x ≈ y → x ≡ y
+  bisim′ : ∀ {x y} → x ≈′ y → x ≡ y
+
+  bisim′ {zero} {zero}  (con p) = p
   bisim′ {zero} {suc x} (con p) = p
   bisim′ {suc x} {zero} (con p) = p
   bisim′ {suc x} {suc y} (con eq) i = suc (bisim eq i)
+  force (bisim eq i) = bisim′ (prove eq) i
+
+  misib : ∀ {x y} → x ≡ y → x ≈ y
+  misib′ : ∀ {x y} → x ≡ y → x ≈′ y
 
-misib : ∀ {x y} → x ≡ y → x ~ y
-prove (misib x≡y) = misib′ (cong force x≡y)
-  where
-  misib′ : ∀ {x y} → x ≡ y → x ~′ y
-  misib′ {zero} {zero} p = con tt
+  misib′ {zero} {zero} p = con p
   misib′ {zero} {suc x} p = con p
   misib′ {suc x} {zero} p = con p
-  misib′ {suc x} {suc y} p = con λ where .prove → misib′ (cong pred′ p)
+  -- misib′ {suc x} {suc y} p = con λ where .prove → misib′ (cong pred′ p)
+  misib′ {suc x} {suc y} p = con (misib (cong pred′′ p))
+  prove (misib x≡y) = misib′ (cong force x≡y)
+
+  iso1 : ∀ {x y} → (p : x ≈ y) → misib (bisim p) ≡ p
+  iso1′ : ∀ {x y} → (p : x ≈′ y) → misib′ (bisim′ p) ≡ p
+
+  iso1′ {zero} {zero} (con p) i = con p
+  iso1′ {zero} {suc x} (con p) i = con p
+  iso1′ {suc x} {zero} (con p) i = con p
+  iso1′ {suc x} {suc y} p i = p
+  prove (iso1 p i) = iso1′ (prove p) i
+
+  iso2 : ∀ {x y} → (p : x ≡ y) → bisim (misib p) ≡ p
+  iso2′ : ∀ {x y} → (p : x ≡ y) → bisim′ (misib′ p) ≡ p
+
+  iso2′ {zero} {zero} p _ = p
+  iso2′ {zero} {suc x} p _ = p
+  iso2′ {suc x} {zero} p _ = p
+  iso2′ {suc x} {suc y} p i j = suc (iso2 (cong pred′′ p) i j)
```

Comment:
Now I'm having a unification error here (CI should show that) and I'm kind of stucking on it.

The target Path should be a Path between `p` and `suc (bisim (misib (cong pred‘’ p)) i)`, if I understand it correctly.

Reply:
It's expanded by the compiler to `p != suc (iso2 (λ i → pred′′ (p i)) j)`